### PR TITLE
added network interface and models

### PR DESCRIPTION
**Pull request by mishraayush89**
review it.

### CODE REVIEW

File: app/src/main/java/anime/stream/shin/base/network/AnimeNetworkprovider.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package anime.stream.shin.base.network
+import anime.stream.shin.base.network.models.AnimeDataModel
+import anime.stream.shin.base.network.models.AnimeEpisodeDataModel
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+
+const val BASE_URL = "https://gogoanime.now.sh/api/v1/";
```
Comment (snehil-shrivastava) [inline app/src/main/java/anime/stream/shin/base/network/AnimeNetworkprovider.kt:9]: abe isko ek Constraints file mein  daal do... bss easy hoga dekhne ke liye

Comment (mishraayush89) [reply app/src/main/java/anime/stream/shin/base/network/AnimeNetworkprovider.kt:9]: ok

File: app/src/main/java/anime/stream/shin/base/network/AnimeNetworkprovider.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+interface AnimeNetworkprovider {
```
Comment (snehil-shrivastava) [inline app/src/main/java/anime/stream/shin/base/network/AnimeNetworkprovider.kt:9]: yeh class ko kuch iss type se krdo,
```

    interface AnimeProviderInterface<T> {
        fun getEpisodesList() : Call<T>
    }
    
    data class AnimeEpisodeDataModel(val name: String)
    
    interface GoGoAnime : AnimeProviderInterface<AnimeEpisodeDataModel>{
    
        @GET("RecentReleaseEpisodes/1")
        override fun getEpisodesList(): Call<AnimeEpisodeDataModel>
    }

```

Kyun ke yeh ek template type generate kr deta hai.. nd baad mein koi dussra provider ko define krne mein easy kr deta hai..

File: app/src/main/java/anime/stream/shin/base/network/models/Server.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package anime.stream.shin.base.network.models
+
+data class Server(
```
Comment (snehil-shrivastava) [inline app/src/main/java/anime/stream/shin/base/network/models/Server.kt:3]: isska use kidhr hai

**Pull request closed without merging**